internal/oci/runtime_vm.go

```diff
@@ -52,6 +53,33 @@ type containerInfo struct {
 	cio *cio.ContainerIO
 }
 
+func buildShimV2Path(path string) string {
+	// shim v2 expects the runtime name to be in the following version:
+	//        ($dir.)?$prefix.$name.$version
+	//        -------- ------ ----- ---------
+	//              |     |     |      |
+	//              v     |     |      |
+	//      "/usr/local/bin"    |      |
+	//        (optional)  |     |      |
+	//                    v     |      |
+	//             "containerd.shim."  |
+	//                          |      |
+	//                          v      |
+	//                     "kata-qemu" |
+	//                                 v
+	//                                "v2"
+	const expectedPrefix = "containerd-shim-"
+	const expectedVersion = "-v2"
+
+	const binaryPrefix = "containerd.shim"
+	const binaryVersion = "v2"
+
+	runtimePrefix := filepath.Dir(path)
+	runtimeName := strings.TrimLeft(strings.TrimRight(filepath.Base(path), expectedVersion), expectedPrefix)
```

Comment:
Oh, I thought Trim{Left,Right} would take prefixes instead of cutsets.  Okay, TIL and this is something I need to update.